Realm/RLMAccessor.mm

```diff
@@ -890,15 +914,17 @@ void RLMSetSwiftPropertyAny(__unsafe_unretained RLMObjectBase *const obj, uint16
 }
 
 id RLMAccessorContext::box(realm::Mixed v) {
-    return RLMMixedToObjc(v, _realm, &_info);
+    auto property = (currentProperty) ? currentProperty : _info.propertyForTableColumn(_colKey);
```

Comment:
For any other mixed wrapped types besides list and dictionary the accessor is constructed without the current property because it is not needed to box the native value